src/scripts/pages/home.js

```diff
@@ -23,15 +22,19 @@ document.addEventListener('DOMContentLoaded', () => {
   }
 
   function showPublicCategories () {
-    $publicCategoriesSection.classList.add('active');
-    $privateCategoriesSection.classList.remove('active');
-    $toggleBtn.innerText = privateBtn;
+    if ($publicCategoriesSection || $privateCategoriesSection) {
+      $publicCategoriesSection.classList.add('active');
+      $privateCategoriesSection.classList.remove('active');
+      $toggleBtn.innerText = privateBtn;
+    }
   }
 
   function showPrivateCategories () {
-    $publicCategoriesSection.classList.remove('active');
-    $privateCategoriesSection.classList.add('active');
-    $toggleBtn.innerText = publicBtn;
+    if ($publicCategoriesSection || $privateCategoriesSection) {
```

Comment:
Same here, we need to check every element we are accessing properties on

```suggestion
    if ($publicCategoriesSection && $privateCategoriesSection && $toggleBtn) {
```